Sources/SQLiteKit/SQLiteConnection+SQLKit.swift

```diff
@@ -47,80 +66,136 @@ internal struct _SQLiteDatabaseVersion: SQLDatabaseReportedVersion {
+        if let stringValue {
+            assert(
+                stringValue.hasPrefix("\(components.major).\(components.minor).\(components.patch)"),
+                "SQLite version string '\(stringValue)' must prefix-match numeric version '\(intValue)'"
+            )
             self.stringValue = stringValue
         } else {
             self.stringValue = "\(components.major).\(components.major).\(components.patch)"
         }
     }
 
-    /// The major version number. This is likely to be 3 for a long time to come yet.
-    var majorVersion: Int { Self.components(of: self.intValue).major }
+    /// The major version number.
+    ///
+    /// This is likely to be 3 for a long time to come yet.
+    var majorVersion: Int {
+        Self.components(of: self.intValue).major
+    }
     
     /// The minor version number.
-    var minorVersion: Int { Self.components(of: self.intValue).minor }
+    var minorVersion: Int {
+        Self.components(of: self.intValue).minor
+    }
     
     /// The patch version number.
-    var patchVersion: Int { Self.components(of: self.intValue).patch }
+    var patchVersion: Int {
+        Self.components(of: self.intValue).patch
+    }
 
-    /// See ``SQLDatabaseReportedVersion/isEqual(to:)``.
+    // See `SQLDatabaseReportedVersion.isEqual(to:)`.
     func isEqual(to otherVersion: any SQLDatabaseReportedVersion) -> Bool {
-        (otherVersion as? _SQLiteDatabaseVersion).map { $0.intValue == self.intValue } ?? false
+        (otherVersion as? Self).map { $0.intValue == self.intValue } ?? false
     }
     
-    /// See ``SQLDatabaseReportedVersion/isOlder(than:)``.
+    // See `SQLDatabaseReportedVersion.isOlder(than:)`.
     func isOlder(than otherVersion: any SQLDatabaseReportedVersion) -> Bool {
-        (otherVersion as? _SQLiteDatabaseVersion).map {
-            (self.majorVersion < $0.majorVersion ? true :
-            (self.majorVersion > $0.majorVersion ? false :
-            (self.minorVersion < $0.minorVersion ? true :
-            (self.minorVersion > $0.minorVersion ? false :
-            (self.patchVersion < $0.patchVersion ? true : false)))))
+        (otherVersion as? Self).map {
+            (self.majorVersion != $0.majorVersion ? self.majorVersion < $0.majorVersion :
+            (self.minorVersion != $0.minorVersion ? self.minorVersion < $0.minorVersion :
+            (self.patchVersion < $0.patchVersion)))
         } ?? false
     }
 }
 
-private struct _SQLiteSQLDatabase: SQLDatabase {
-    let database: any SQLiteDatabase
+/// Wraps a `SQLiteDatabase` with the `SQLDatabase` protocol.
+private struct SQLiteSQLDatabase<D: SQLiteDatabase>: SQLDatabase {
+    /// A trivial wrapper type to work around Sendable warnings due to SQLiteNIO not being Sendable-correct.
+    struct FakeSendable<T>: @unchecked Sendable {
+        let value: T
+    }
+    
+    /// The underlying database.
+    let database: FakeSendable<D>
```

Comment:
I'm going to push back on this - `SQLiteDatabase` can easily be sendable, as can the types that conform to it so I think we should do that

Reply:
Working on that over in the SQLiteNIO PR, but I didn't want to block this one on that.

Reply:
IMO it makes sense to get the other PR landed first with the sendable annotations and then land this. Don't feel super strongly about it though

Reply:
Yeah, that's probably better at this point. I didn't even have the SQLiteNIO PR in mind yet when I first wrote this one.